modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -1143,10 +1146,11 @@ def date_range_overlap?(range_one, range_two)
       end
 
       # Will check for a real date including leap year
-      def date_is_valid?(date, property)
+      def date_is_valid?(date, property, is_full_date = false) # rubocop:disable Style/OptionalBooleanParameter
         return if date.blank?
 
         raise_date_error(date, property) unless /^[\d-]+$/ =~ date # check for something like 'July 2017'
+        return false if is_full_date && !date.match(YYYY_MM_DD_REGEX)
         return true if date.match(YYYY_YYYYMM_REGEX) # valid YYYY or YYYY-MM date
 
         date_y, date_m, date_d = date.split('-').map(&:to_i)
```

Comment:
Is this method doing too much?

1. Could we pass in a regex instead of a boolean flag, so we can support different date formats in the future?
2. Could we reduce the other checks to just a regex match? E.g.,
   ```rb
   def date_is_valid?(date, property, regex = YYYY_YYYYMM_REGEX)
     return true if date.match(regex)
     raise_date_error(date, property)
   end
   ```
 
   In other words, what purpose do `return if date.blank?` and `raise_date_error(date, property) unless /^[\d-]+$/ =~ date` serve? 
   
3. Do we want the `Date.valid_date?` check? If so, should we ensure it runs every time? Currently, we return early based on the regex check.

Reply:
Never mind, I see we run through the `Date.valid_date?` checks when we have a YYYY-MM-DD date.